tests/phpunit/utils/trait-upe-utils.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/**
+ * Provides methods useful when testing UPE-related logic.
+ */
+trait UPE_Utils {
```

Comment:
Good idea! Should this be `UPE_Test_Utils`, to ensure that it's only used in tests?
Or maybe now that I think about it, having it in the `tests` directory is sufficient 🤷